pkg/types/types.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"database/sql"
+	"time"
+)
+
+type ProviderCluster struct {
+	ClusterID   string
+	AccountID   string
+	SatelliteID string
+	MetaData    sql.NullString
+	Spec        sql.NullString
+	Status      sql.NullString
+}
+
+type Request struct {
+	Name string `json:"name"`
+}
+
+type Result struct {
+	Requeue bool
+	After   time.Duration
+}
```

Comment:
These types are part of the formen interface. They are not data types.
Lets keep them in the Fromen package

Reply:
i kept it here to avoid cycle import problem, this is also required in db folder to send the notifications and formen is also importing db package

Reply:
That should not be a problem the DB package should not relay on these at all

Reply:
db also need to setup the notification as you suggested and for that we need this Request struct to be imported.